pkg/connector/users.go

```diff
@@ -21,40 +24,114 @@ func (o *userBuilder) ResourceType(ctx context.Context) *v2.ResourceType {
+		return nil, "", outputAnnotations, fmt.Errorf("listing employees failed: %w", err)
 	}
-	nextPage = makeNextPageStr(offset, limit, resp.TotalCount)
 
-	users, err := employees2users(resp.Employees, parentResourceID)
-	if err != nil {
-		return nil, "", nil, err
+	var resources []*v2.Resource
+	for _, employee := range positions {
+		userResource, err := parseUserToResource(employee, parentResourceID)
+		if err != nil {
+			return nil, "", outputAnnotations, err
+		}
+		resources = append(resources, userResource)
 	}
 
-	return users, nextPage, nil, nil
+	return resources, nextPageToken, outputAnnotations, nil
 }
 
 // Entitlements always returns an empty slice for users.
 func (o *userBuilder) Entitlements(_ context.Context, resource *v2.Resource, _ *pagination.Token) ([]*v2.Entitlement, string, annotations.Annotations, error) {
 	return nil, "", nil, nil
 }
 
-// Grants always returns an empty slice for users since they don't have any entitlements.
+// The Grants function for positions (roles) is implemented here on the userBuilder for performance reasons.
+// This allows assigning grants directly while iterating through users, as the Paylocity API
+// already includes the 'positionCode' in the employee list, avoiding the need to iterate
+// over all positions to resolve each user's assignment individually.
 func (o *userBuilder) Grants(ctx context.Context, resource *v2.Resource, pToken *pagination.Token) ([]*v2.Grant, string, annotations.Annotations, error) {
-	return nil, "", nil, nil
+	userId := resource.Id.Resource
+	outputAnnotations := annotations.New()
+
+	user, rateLimitDesc, err := o.service.GetUserById(ctx, userId)
+	if rateLimitDesc != nil {
+		outputAnnotations.WithRateLimiting(rateLimitDesc)
+	}
+	if err != nil {
+		return nil, "", outputAnnotations, fmt.Errorf("failed to get user %s for grants: %w", userId, err)
+	}
+
+	var grants []*v2.Grant
+	if user.Position.PositionCode != "" {
+		positionResource := &v2.Resource{
+			Id: &v2.ResourceId{
+				ResourceType: roleResourceType.Id,
+				Resource:     user.Position.PositionCode,
+			},
+		}
+		grants = append(grants, grant.NewGrant(positionResource, PermissionMember, resource.Id))
+	}
+
+	return grants, "", outputAnnotations, nil
 }
 
-func newUserBuilder(client *client.Client) *userBuilder {
-	return &userBuilder{c: client}
+func parseUserToResource(user *client.User, parentResourceID *v2.ResourceId) (*v2.Resource, error) {
+	profile := map[string]interface{}{
+		"user_id":       user.ID,
+		"display_name":  user.DisplayName,
+		"first_name":    user.Info.FirstName,
+		"last_name":     user.Info.LastName,
+		"job_title":     user.Info.JobTitle,
+		"employee_type": user.Position.EmployeeType,
+		"position_code": user.Position.PositionCode,
+		"department":    user.Position.Department,
+	}
+
+	var status v2.UserTrait_Status_Status
+	normalizedStatus := strings.ToLower(user.Status)
+	switch normalizedStatus {
+	case "Active":
```

Comment:
strings.ToLower result will never match "Active" as the bot said. When you test this user, do all the users return as disabled?